common/cuda_hip/base/unaligned_access.hpp.inc

```diff
@@ -0,0 +1,137 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2023, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+
+/**
+ * Copies value of type T starting at byte start from ptr.
+ *
+ * @tparam T     the type of value
+ *
+ * @param ptr    the starting pointer
+ * @param start  the offset
```

Comment:
I have not been brilliant with these names ...

My suggestion would be:
-  Use "displ" instead of "start"
- Add "(in bytes)" at the end of the explanation

===============================
I do not know, why I cannot add comments related to include/ginkgo/core/matrix/bccoo.hpp ...

My suggestions would be:
- I will change: rows, cols, offsets, chunk
- I know, types is only used for block compression variant, but I would prefer to change it to "compression_types"
- block_size is not changed in each "chunk", except for the last one if it is not big enough. Therefore, I think it could be confusing to use "block_chunk".

Thanks in advance!!

Reply:
About block_size/chunck_size, I forgot to mention that it contains the number of nonzeros elements in each block, not the number of bytes in each chunck

Reply:
I understand your point about chunk_size.
But for me, chunk is more related to bytes, and the size of each chunk is different.
Obviously, using element and block compression makes it difficult to use of block_size difficult, and changing the compression types would solve the problem.

First, I will justify the current names:
* Bccoo format divides a matrix in coo format into blocks, whose size is block_size. 
* In "element variant", the analysis of the applied compression is done element by element, 
* In "block variant", the compression applied is the least restrictive of those decided for all the elements of the block. 
As a consequence, I consider that:
* block_size should be kept, because it is a common parameter for all bccoo formats.
* "element variant" could be kept or moved to "individual variant", since each nonzeros is analysed separately.
* "block variant" could be moved to "group variant", since the chosen compression  is the least restrictive for a group of nonzeros.  

About "displ" or "dspl", it tries to be a short for "displacement!

Reply:
In that case feel free to keep the name `block_size`, just make sure that any user understands what it actually does.

Reply:
In summary, the rename in include/ginkgo/core/matrix/bccoo.hpp would be:
- rows -> start_rows
- cols -> start_cols
- types -> compression_types
- chunk -> data
- offsets -> data_offsets
- "element variant" -> "element variant" or "element variant" -> "individual variant", which do you prefer?
- "block variant" -> "group variant"  

Of course the names in unaligned_access files will be changed:
- Using "dspl" instead of "start"
- Adding "(in bytes)" to the end of the explanation

I think these renames will help users to understand how the format works.

Please, confirm if you agree with these changes. 

Thanks in advance!!!!

Reply:
Sounds good!
I have a slight preference for `individual variant`, but both sound good to me.

Reply:
Perfect!! Thank you!!

Reply:
I have completed the changes in the code and they have been tested in OMP and CUDA, but still not in HIP and DPCPP. A summary of the changes appears at the end.
Please, confirm that you agree to the changes. Thanks in advance!!

=======================

1. Changes in unaligned files

2. Changes related to chunk
include/ginkgo/core/matrix/bccoo.hpp
* chunk -> compressed_data
Rest of files
* Variables 
   chunk_data -> compressed_data
* Methods
   _chunk -> _compressed_data

3. Changes related to Bccoo vectors
include/ginkgo/core/matrix/bccoo.hpp
* rows -> start_rows
* cols -> start_cols
* types -> compression_types
* offsets -> block_offsets
Rest of files
* rows_data -> start_rows
* cols_data -> start_cols
* types_data -> compression_types
* offsets_data -> block_offsets

4. Changing name of Bccoo variants
All files:
* uint8 ind -> uint8 key
* element —> individual , elm —> ind  , Elm —> Ind
* block —> group , blk —> grp , Blk —> Grp